### PR TITLE
Add compiler warning for map.getValue and add docstring

### LINKED ISSUE #16065: Update doc for map.getValue(key)
`.getValue()` was added to support different class types as keys/values in maps. This issue is a gentle reminder for adding a doc-string to the signature of the proc.

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
Adjust `map.getValue` so that it uses an embedded `compilerError` instead of a where clause. The error fires if `map.getValue` is called when the map value type is not a non-nilable class. Add another `compilerError` that fires if the map value type is an owned class. The error tells the user to call `getBorrowed` instead. Add a docstring describing the behavior of `map.getValue`. Add three tests to lock in the intended behavior of `map.getValue`.

Resolves #16065.

---

Testing:

- [x] `ALL` on `linux64` when `CHPL_COMM=none`
- [x] `library/standard/Set` on `darwin` when `CHPL_COMM=none`, with futures
- [x] Updated docs rendered locally

### CODE REVIEW

File: test/library/standard/Map/testGetValue.chpl [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+class C {
+  var i: int;
+}
+
+var m: map(string, shared C);
+
+m.add("one", new shared C(1));
+m.add("two", new shared C(2));
+
+var x = m.getValue("one");
```
Comment (daviditen) [inline test/library/standard/Map/testGetValue.chpl:12]: This seems to show that something isn't working right. `getValue` returns `const`, but this stores it into a `var` and then modifies it.

File: modules/standard/Map.chpl [hunk 2]
```diff
@@ -329,8 +329,14 @@ module Map {
       return result;
     }
 
-    proc getValue(k: keyType) const
-    where isNonNilableClass(valType) {
+    /* Get a copy of the element stored at position `k`. This method is only
+       available when a map `valType` is a non-nilable class.
```
Comment (daviditen) [inline modules/standard/Map.chpl:333]: "a map's"?

File: test/library/standard/Map/testGetValue.chpl (hunk 1 above)
Comment (dlongnecke-cray) [reply test/library/standard/Map/testGetValue.chpl:12]: Do we even document the `const` return intent today? I am going to go out on a limb here and assume it means "const value" and not shorthand for "const ref"...

If that's the case I would expect the compiler to make another copy when moving the tmp at the callsite into the variable.

I'll go ahead and add a test to verify that.

File: test/library/standard/Map/testGetValueBadValType.chpl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+use Map;
+
+class C {
```
[the remark this answers is not in the document]
Comment (dlongnecke-cray) [reply test/library/standard/Map/testGetValueBadValType.chpl:3]: Thanks!

File: modules/standard/Map.chpl (hunk 2 above)
Comment (dlongnecke-cray) [reply modules/standard/Map.chpl:333]: Thanks!

**Comment by dlongnecke-cray** [comment]
To be honest, I was wondering that as I wrote the code. At the time though, I was just looking to correct a mistake instead of fiddle with the semantics.

Technically though, the only restriction of a method like `getValue` is that the value type is copyable.

**Pull request merged**